src/app/stores/Reducers.js

```diff
@@ -23,35 +24,50 @@ export const workDetail = (state = null, action) => {
   }
 };
 
-export const query = (state = null, action) => (
-  (action.payload) ?
-    action.payload.query :
-    state
-);
+export const searchQuery = (state = null, action) => {
+  if (action.type === Actions.SET_QUERY) {
+    return action.searchQuery;
+  }
 
-export const field = (state = null, action) => (
-  (action.payload) ?
-    action.payload.field :
-    state
-);
+  return state;
+};
 
-export const allowedFilters = (state = null, action) => (
-  (action.payload) ?
-    action.payload.allowedFilters :
-    state
-);
+export const searchField = (state = null, action) => {
+  if (action.type === Actions.SET_FIELD) {
+    return action.searchField;
+  }
+
+  return state;
+};
 
 export const sort = (state = null, action) => (
-  (action.payload) ?
-    action.payload.sort :
+  (action.sort) ?
+    action.sort :
     state
 );
 
-export default combineReducers({
+// export const resetSearch = (state = null, action) => {
+//   if (action.type === Actions.RESET_SEARCH) {
+//     return initialState;
+//   }
+
+//   return state;
+// };
+
+const appReducer = combineReducers({
   searchResults,
-  query,
-  field,
-  allowedFilters,
+  searchQuery,
+  searchField,
   sort,
   workDetail,
 });
+
+export const rootReducer = (state, action) => {
+  if (action.type === Actions.RESET_SEARCH) {
+    state = initialState;
+  }
+
+  return appReducer(state, action);
+};
+
+export default rootReducer;
```

Comment:
Here you are exporting `rootReducer` twice, once as the default and once as a variable. Right now you are importing it as `import searchReducer from './Reducers';` which would be this `default` export. But you can also do `import { rootReducer } from './Reducers';` and the app would work just the same. For consistency and to reduce confusion, pick one or the other.

The default option may be the best one since you currently only do the former import.

Reply:
Changing to reflect the reducer name change.